HHS/OPRE-OPS
# Ops 1960/agreement patch bug

## What changed

- adds custom hook to check state change 
- fixes the agreement tabs to not prompt for leaving page only if changes are detected

## Issue

#1960 

## How to test

1. Goto Agreement Details
2. Enter `Edit Mode`
3. Don't make any changes to Agreement meta data
4. Goto Budget lines tab
5. 🟢 should allow you
6. Make edits to the Agreement meta data
7. Attempt to go to Budget lines tab without saving changes
8. 🔴 should warn you in via confirmation modal

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
